entry_types/scrolled/package/src/frontend/MediaPlayer/index.js

```diff
@@ -0,0 +1,44 @@
+import React, {useEffect, useContext} from 'react';
+import PlayerContainer from './PlayerContainer';
+import ScrollToSectionContext from "../ScrollToSectionContext";
+import {useMediaSettings} from '../useMediaSettings';
+
+export function MediaPlayer(props){
+  let player = undefined;
+  let scrollToSection = useContext(ScrollToSectionContext);
+  let mediaSettings = useMediaSettings();
+
+  let onSetup = (newPlayer)=>{
+    player = newPlayer;
+  }
+  let onDispose = ()=>{
+    player = undefined;
+  }
+
+  useEffect( () => {
+    if (player && mediaSettings.mediaOff === false && props.autoplay !== false) {
+      if (props.state === 'active') {
+        if (player.readyState() > 0) {
+          player.play();
+        } else {
+          player.on('loadedmetadata', player.play);
+        }
+      } else {
+        player.pause();
+      }
+    }
+  }, [mediaSettings.mediaOff, props.autoplay, props.state, player]);
```

Comment:
Using a local variable for `player` does not work: On first render, the `onSetup` function and the function passed to `useEffect` share the closure of the render function. So when the effect fires it might be able to read the `player` variable set by `onSetup`. On next render, `onSetup` will not be executed again, though. Still the effect function now sees the closure of this new call of the render function. `player` will thus be undefined. To fix this, `player` needs to be a ref. As a rule of thumb, whenever you would have [assigned something to `this`](https://github.com/codevise/pageflow/blob/023fdb7c5b3b917bb94f4adafa18a31f8bd9c753/entry_types/paged/packages/pageflow-paged-react/src/media/components/createFilePlayer/index.jsx#L46) on a class component, you will probably need a ref in a function component.

Also `mediaOff` will only ever be `true` or `undefined`. The strict equality check will therefore always fail. Checking for falsiness (i.e. `!mediaSettings.mediaOff`) makes autoplay work.

Normally I'd say let's add a test to cover this logic. But I'm hesitant since this functionality will change when we wire up `playerState` in the next step. It might be a good example case, though, since a lot of similar specs will be needed for the `playerState` integration.